odxtools/externflashdata.py

```diff
@@ -0,0 +1,34 @@
+# SPDX-License-Identifier: MIT
+from dataclasses import dataclass
+from typing import Any
+from xml.etree import ElementTree
+
+from .datafile import Datafile
+from .exceptions import odxrequire
+from .flashdata import Flashdata
+from .odxdoccontext import OdxDocContext
+from .odxlink import OdxLinkDatabase, OdxLinkId
+from .snrefcontext import SnRefContext
+from .utils import dataclass_fields_asdict
+
+
+@dataclass(kw_only=True)
+class ExternFlashdata(Flashdata):
+    datafile: Datafile
+
+    @staticmethod
+    def from_et(et_element: ElementTree.Element, context: OdxDocContext) -> "ExternFlashdata":
+        kwargs = dataclass_fields_asdict(Flashdata.from_et(et_element, context))
+
+        datafile = Datafile.from_et(odxrequire(et_element.find("DATAFILE")), context)
+
+        return ExternFlashdata(datafile=datafile, **kwargs)
+
+    def _build_odxlinks(self) -> dict[OdxLinkId, Any]:
+        return super()._build_odxlinks()
```

Comment:
Why overload a function if you are just going to call super?

Reply:
basically to signal that these methods ought to be called by the parent object. That said, this is not consistently done in the code base...

Reply:
I don't get what you mean

Reply:
to allow easy extension of these classes or changes in the class hierarchy. (IMO, the fact that in OOP all methods of parent classes are implicitly available in derived classes is a course and a benefit at the same time...)

Reply:
(in the original comment, I should have better said "owner object" instead of "parent object"...)